packages/web/src/modules/zoom/components/zoom-provider.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useCallback, useState, ChangeEvent } from 'react';
+import { hooks, ui } from '@application';
+import { Components } from '@types';
+
+import { zoomAuthorize, generateSignature } from '../requests';
+import { ZoomContext, ZoomContextType, Role } from '../context';
+import { authorizeZoom } from '../services';
+import ZoomActivityView from './zoom-activity-view';
+
+const { useApi, useQuery } = hooks;
+const { Button, Container, Input, Row, Col, Spinner } = ui;
+
+const ZoomProvider: Components['ZoomProvider'] = ({ redirectUri }) => {
+  const [zoomContextState, setZoomContextState] = useState<ZoomContextType>({
+    userSignature: '',
+    hostUserZakToken: null,
+    meetingNumber: '',
+    username: '',
+    password: '',
+    role: null,
+  });
+
+  const { code } = useQuery<{ code?: string; path?: string }>();
+  const zoomAuthorizeApi = useApi(zoomAuthorize);
+  const zoomSignatureApi = useApi(generateSignature);
+
+  useEffect(() => {
+    if (
+      code &&
```

Comment:
Move this condition to a function. Maybe to `utils` in `zoom` module.

Or a variable.

Will be more descriptive.